OpenRA.Mods.Common/Activities/Air/FlyCircle.cs

```diff
@@ -29,6 +29,9 @@ public FlyCircle(Actor self, int ticks = -1, int turnSpeedOverride = -1)
 
 		public override Activity Tick(Actor self)
 		{
+			if (NextActivity != null && remainingTicks < 0)
```

Comment:
If you make this `remainingTicks <= 0` then you can drop the redundant `else if (remainingTicks == 0)` line below.